### DISCUSSION TITLE
custom regex for public github dependency with private endpoint

### DISCUSSION THREAD

**Discussion by MikeDevresse** [renovatebot/renovate#25910] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.64.2

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### What is your question?

Hi,
I have a kubernetes cluster with an argocd instance deployed. I want to scan a Kustomize file to update my argocd version. Here is my file:
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
  - https://raw.githubusercontent.com/argoproj/argo-cd/v2.8.4/manifests/install.yaml
  - ...

patches:
  - ...
```
As you can see I just need to get the argoproj/argo-cd as depName, and v2.8.4 as currentValue. So I came up with the following configuration:
```
customManagers: [
        {
            customType: "regex",
            fileMatch: ["\\.ya?ml$"],
            matchStrings: ["https:\\/\\/raw\\.githubusercontent\\.com\\/(?<depName>[a-z\\-]+\\/[a-z\\-]+)\\/(?<currentValue>[^\\/]+)",],
            datasourceTemplate: "github-releases"
        }
    ]
```
But unfortunately, I get the following error:
```
 WARN: GitHub token is required for some dependencies (repository=Platform/ArgoCD)
       "githubDeps": ["argoproj/argo-cd"]
```
I tried changing datasource to be github-tags but I got the same error.

I think it has something to do with the fact that I have an endpoint and token parameter which makes the github datasource believe that I want to search on this endpoint with this token but those parameters are my Azure devops auth settings for creating PRs.

Thanks for your help in advance !

**Comment by rarkins** [comment]
The problem will go away if you add a github.com PAT. See GITHUB_COM_TOKEN

**Reply by MikeDevresse** [reply]
And is it possible to not specify any as it is a public repository ? If I can get away without specifying a github repository will be great

**Reply by rarkins** [reply]
Renovate uses the graphql APIs which have mandatory authentication

**Reply by MikeDevresse** [reply, marked as answer]
I managed to do it with a custom datasource:

```
    customManagers: [
        {
            customType: "regex",
            fileMatch: ["\\.ya?ml$"],
            matchStrings: ["https:\\/\\/raw\\.githubusercontent\\.com\\/(?<depName>[a-z\\-]+\\/[a-z\\-]+)\\/(?<currentValue>[^\\/]+)",],
            datasourceTemplate: "custom.github",
        }
    ],
    customDatasources: {
        github: {
            defaultRegistryUrlTemplate: "https://api.github.com/repos/{{packageName}}/git/refs/tags",
                transformTemplates: [
                "{\"releases\": $.ref @ $myRef #$i.{\"version\": $myRef.$substringAfter(\"refs/tags/\")}}"
            ]
        }
    }
```

Thanks for your help !
